--- configeth.orig      2025-05-29 04:33:59.116420802 +0000
+++ configeth   2025-05-29 04:36:07.663941690 +0000
@@ -763,7 +763,22 @@
                 tmp_con_name=${str_inst_nic}"-tmp"
                 nmcli con modify $con_name connection.id $tmp_con_name
             fi
+
+           str_inst_gateway=${MASTER}
+           echo "GATEWAY is ${str_inst_gateway}" >> /var/log/xcat/xcat.log
             nmcli con add type ethernet con-name $con_name ifname ${str_inst_nic} ipv4.method manual ipv4.addresses  ${str_inst_ip}/${str_inst_prefix} connection.autoconnect-priority 9 gw4 ${str_inst_gateway}
+            default_routes=$(ip route show default | wc -l)
+
+            # If there are multiple default routes, remove the one for the specified NIC
+            if [ "$default_routes" -gt 1 ]; then
+                # Get the active connection name for the admin NIC
+                echo "Setting never-default to yes for $con_name" >> /var/log/xcat/xcat.log
+                nmcli connection modify "$con_name" ipv4.never-default yes
+                nmcli con reload
+                nmcli con up $con_name
+            else
+                log_info "Only one default route found, no action needed."
+            fi
             str_conf_file_1="/etc/sysconfig/network-scripts/ifcfg-xcat-${str_inst_nic}-1"
             if [ -f $str_conf_file_1 ]; then
                 grep $con_name $str_conf_file_1 >/dev/null 2>/dev/null
